third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -214,14 +209,15 @@ static void moveUpTranspose(triton::FuncOp funcOp) {
 }
 
 // Schedule global load and local store ops for better GEMM performance.
-static void scheduleGlobalLoadLocalStore(triton::FuncOp funcOp) {
+static void scheduleGlobalLoadLocalStore(scf::ForOp forOp) {
```

Comment:
This may have implications for other workloads, where it is beneficial to apply outside of for loops. Let's keep it as before.

Reply:
@sjw36, this function is only executed when we are dealing with `pureMatmulProblems`: https://github.com/triton-lang/triton/blob/e2dc77b87be71f4ccb7ad6d3627b5be24407b6b2/third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp#L380-L383